### ISSUE TITLE
Issue when layering tfrmts 

### ISSUE THREAD

**Issue by statasaurus** [GSK-Biostatistics/tlang#125]
```r

demog_example <- function(){
  tfrmt(
  # specify columns in the data
  group = c(rowlbl1,grp),
  label = rowlbl2,
  column = column,
  param = param,
  values = value,
  sorting_cols = c(ord1, ord2),
  
  # Specify body plan
  body_plan = body_plan(
    frmt_structure(group_val = ".default", label_val = ".default", frmt_combine("{n} {pct}",
                                                                                n = integer_frmt,
                                                                                pct = pct_frmt())),
    frmt_structure(group_val = ".default", label_val = "n", integer_frmt),
    frmt_structure(group_val = ".default", label_val = c("Mean", "Median", "Min","Max"), two_sig_fig_frmt),
    frmt_structure(group_val = ".default", label_val = "SD", two_sig_fig_frmt),
    frmt_structure(group_val = ".default", label_val = ".default", p = frmt("")),
    frmt_structure(group_val = ".default", label_val = c("n","<65 yrs","<12 months","<25"), p = pval_frmt)
  ),
  
  # Specify row group plan
  row_grp_plan = row_grp_plan(
    row_grp_structure(group_val = ".default", element_block(post_space = " ")),
    label_loc = element_row_grp_loc(location = "column")
  ),
  
  # Specify column alignment plan
  col_align = col_align_plan(
    element_align(align = c(".",","," "), col = vars(everything()))
  ),
  
  # remove extra cols
  col_plan = col_plan(-grp,
                      -starts_with("ord") )
) 
}

demog_example() %>% 
  tfrmt(
    title = "This is my new dataset",
    footer = "*We did some ANOVAS"
  ) %>% 
  print_to_gt(demog_data)
```
![image](https://user-images.githubusercontent.com/55274484/172830240-9068c343-ac13-4f29-b264-2a426677d134.png)

**Comment by thebioengineer** [comment]
this code won't run b/c its missing the defined frmts, but the root cause of the error you are seeing is resolved in #126. see some fixed code below:

```r

demog_example <- function(tfrmt_obj, sigfigs = 2){

  basic_frmt <- frmt("xxx.xx")
  pct_frmt <- frmt_when("==100" ~ "","==0" ~ "",TRUE ~ frmt("(xx.x %)"))
  pval_frmt <- frmt_when(">0.99" ~ ">0.99","<0.001" ~ "<0.001",TRUE ~ frmt("x.xxx", missing = ""))
  integer_frmt <- frmt("xxx")
  n_pct_frmt <- frmt_combine("{n} {pct}",n = integer_frmt, pct = pct_frmt)
  
  sig_fig_frmt <- function(sigfigs = 2) {frmt(paste0("xxx.",paste0(rep("x", sigfigs),collapse="")))}

  demog_template <- tfrmt(
    # specify columns in the data
    group = c(rowlbl1,grp),
    label = rowlbl2,
    column = column,
    param = param,
    values = value,
    sorting_cols = c(ord1, ord2),
    
    # Specify body plan
    body_plan = body_plan(
      frmt_structure(group_val = ".default", label_val = ".default", sig_fig_frmt(sigfigs)),
      frmt_structure(group_val = ".default", label_val = ".default", n_pct_frmt),
      frmt_structure(group_val = ".default", label_val = "n", integer_frmt),
      frmt_structure(group_val = ".default", label_val = c("Mean", "Median", "Min","Max"), sig_fig_frmt(sigfigs)),
      frmt_structure(group_val = ".default", label_val = "SD", sig_fig_frmt(sigfigs)),
      frmt_structure(group_val = ".default", label_val = ".default", p = frmt("")),
      frmt_structure(group_val = ".default", label_val = c("n","<65 yrs","<12 months","<25"), p = pval_frmt)
    ),
    
    # Specify row group plan
    row_grp_plan = row_grp_plan(
      row_grp_structure(group_val = ".default", element_block(post_space = " ")),
      label_loc = element_row_grp_loc(location = "column")
    ),
    
    # Specify column alignment plan
    col_align = col_align_plan(
      element_align(align = c(".",","," "), col = vars(everything()))
    ),
    
    # remove extra cols
    col_plan = col_plan(-grp,
                        -starts_with("ord") )
  ) 
  
  
  if(!missing(tfrmt_obj)){
    demog_template <- layer_tfrmt(tfrmt_obj, demog_template)
  }
  
  demog_template
  
}

demog_example() %>% 
  tfrmt(
    title = "This is my new dataset",
    footer = "*We did some ANOVAS"
  ) %>%  
  print_to_gt(demog_data)

```

**Comment by statasaurus** [comment]
Yes sorry I defied those above this but of code

**Comment by thebioengineer** [comment]
resolved with merge in of #126
